### DISCUSSION TITLE
Get 'AP SSID' in bash for pi TFT display

### DISCUSSION THREAD

**Discussion by done7k** [RaspAP/raspap-webgui#1247] [Q&A] [answered]
I have managed to wire another mini TFT display (240x240, 7 pin based on ST7789) to my RPiZeroW and adapted default `stats.py` to work with another library. Since the display is slightly bigger I want to display additionally:
- AP SSID
- number of connected devices to AP

I am not good in bash, As such, could someone to explain me what bash command would allow me to collect such info?

**Comment by billz** [comment, marked as answer]
Good question. You can fetch the SSID like so (substitute `wlan0` for your AP interface, if different):

```
iw dev wlan0 info | grep -Po '(?<=ssid ).*'
```

For the number of connected AP clients, a count of the active leases will do the trick:

```
cat /var/lib/misc/dnsmasq.leases | wc -l 
```

**Reply by done7k** [reply]
Thanks! 
Used following to get my SSID
 `iw dev | grep -C 3 "type AP" | grep -Po '(?<=ssid ).*' `

**Reply by billz** [reply]
That's a good solution. It's a better approach if you have more than one adapter that could host an AP. 
Feel free to share a photo of your finished display, if you like.

**Comment by done7k** [comment]
I noticed the `cat /var/lib/misc/dnsmasq.leases` shows me one dead connection and in total two rows. However on the RaspAP dashboard there is only one live device.

**Reply by billz** [reply]
Yes, leases will persist until they expire. These are shown on RaspAP's **DHCP Server > Client list**. 
A slightly different technique is used to display active client leases on the dashboard:


https://github.com/RaspAP/raspap-webgui/blob/a87d5829f3d42e72f16bf5365483033ee486e9ec/includes/dashboard.php#L193

This takes `dnsmasq.leases` and compares it with the output of `iw dev wlan0 station dump`. eg.:

```
Station f4:30:8b:35:21:41 (on wlan0)
	inactive time:	1000 ms
	rx bytes:	80001
	rx packets:	478
...
```

The `$MACPattern` variable is defined [here](https://github.com/RaspAP/raspap-webgui/blob/a87d5829f3d42e72f16bf5365483033ee486e9ec/includes/dashboard.php#L186).

**Comment by done7k** [comment]
Almost finished the screens. I only hardcoded the app version I am running on - 2.8.5. is there a way to query any file for version id? And would be cool to get some flag if a new version available. is it possible?

**Reply by billz** [reply]
Sure, it's possible. Get the installed app version by reading `RASPI_VERSION` from `includes/defaults.php`:

```
INSTALLED=$(grep -Po "^\s+'RASPI_VERSION' => '\K.*?(?=')" includes/defaults.php)
echo $INSTALLED 
2.8.5
```

Fetch the latest release from the GitHub API like so:

```
LATEST=$(curl -s "https://api.github.com/repos/RaspAP/raspap-webgui/releases/latest" | grep -Po '"tag_name": "\K.*?(?=")' )
echo $LATEST 
2.8.6
```

By comparing these values you could indicate that a new release is available.

**Comment by done7k** [comment]
I will leave pictures here..

https://user-images.githubusercontent.com/55544417/198727558-0f0f0c78-a782-451f-a701-fce512e28c86.mp4

![RaspAP-screen-0b](https://user-images.githubusercontent.com/55544417/198729357-4b6db4c5-cab9-4fee-a5c4-a155037a8ab2.jpg)
![RaspAP-screen-1b](https://user-images.githubusercontent.com/55544417/198729399-630eeb23-b28c-413c-8090-5bebf9d241ad.jpg)
